lib/fog/orchestration/openstack/requests/create_stack.rb

```diff
@@ -28,6 +30,19 @@ def create_stack(arg1, arg2 = nil)
             }.merge(arg2.nil? ? {} : arg2)
           end
 
+          # Templates should always:
+          #  - be strings
+          #  - contain URI references instead of relative paths.
+          # Passing :template_url may not work well with `get_files` and remote `type`:
+          #  the python client implementation in shade retrieves from :template_uri
+          #  and replaces it with :template.
+          #  see https://github.com/openstack-infra/shade/blob/master/shade/openstackcloud.py#L1201
+          #  see https://developer.openstack.org/api-ref/orchestration/v1/index.html#create-stack
+          hot_resolver = Fog::Orchestration::Util::RecursiveHotFileLoader.new(options[:template] || options[:template_url])
```

Comment:
suggest `file_resolver` to better describe the purpose of the variable

Reply:
Use the resolver only when `options[:files]` does not exist. In other words, no need to load files if the caller already provides them. Update the comments in https://github.com/fog/fog-openstack/pull/317/files#diff-360fd267b1a87d726bdd369c23976486R10

Reply:
@bzwei in shade they merge passed and retrieved :files. Without the retrieved ones the stack will fail.